### DISCUSSION TITLE
Can't delete an Inventory created with wrong `ssh_key_id`. Even with Admin

### DISCUSSION THREAD

**Discussion by esturniolo** [semaphoreui/semaphore#1840] [Q&A] [answered]
Hi all!

I created a new inventory using the API and I made a little mistake: put a `ssh_key_id` not existent in that project.
I forgot to change it and I use 1 like the examples.

The problem now is that I've a 404 every time that I try to delete it or change it 
```
HTTP/1.1 404 Not Found
Content-Type: application/json
Date: Mon, 18 Mar 2024 12:56:43 GMT
Content-Length: 0
```
![image](https://github.com/ansible-semaphore/semaphore/assets/5169045/a4c60a67-7953-4c5b-9f5e-75e84e8b5bf9)

I tried using the admin user without any success.

Any idea?

Thanks in advance!

**Comment by fiftin** [comment]
Hi @esturniolo What database do you use? 
Looks like it is BoltDB because MySQL and Postgres has consistency check on database level.

**Reply by esturniolo** [reply]
Hi @fiftin 
We're using AWS RDS.

**Reply by fiftin** [reply, marked as answer]
It is a bug. Thanks a lot. Will be fixed ASAP.

**Reply by esturniolo** [reply]
Thanks!!!!

**Reply by fiftin** [reply]
@esturniolo 
I've added the fix that will prevent this from happening. But this will not fix old mistakes. 
Do you have access to the database? Will you be able to manually delete the inventory?

**Reply by esturniolo** [reply]
May be.
There is some advice in case to delete this manually?

**Reply by fiftin** [reply]
```sql
delete from project__inventory where id = your_indentory_id
```

**Comment by IvoARL** [comment]
Hey guys, i have the same problem here but HTTP Post Request 

<img width="773" height="817" alt="image" src="https://github.com/user-attachments/assets/7373925f-7abf-42fa-baef-dc1ed80f53a3" />

<img width="1275" height="811" alt="image" src="https://github.com/user-attachments/assets/3ffc9e20-f978-4e28-8506-59f265d8eec1" />


now for project 1

<img width="1221" height="766" alt="image" src="https://github.com/user-attachments/assets/bc41c7f7-5420-4f5c-82bc-2cfe3112ee11" />
<img width="1279" height="816" alt="image" src="https://github.com/user-attachments/assets/b41c3900-0449-43d4-9f47-aa18936e2bda" />


so tell me what can i do here? i wanna do a request http post to create repo for each project using ansible scripts. :C

**Reply by IvoARL** [reply]
how to know the ssh_key_id? is it really needed?

**Reply by fiftin** [reply]
@IvoARL it looks correct. Yes, it is needed. Could you show the response body, it can contains error details.

**Reply by IvoARL** [reply]
Well i dont think there is much of the response. It only says request resource could not be found. Its strange this ssh_key_id since the ansible scripts are on a directory inside the container running semaphore. Why is it requeired? I realise that while i was testing the scripts setuping automaticly the semaphore for diferent projects the id i usualy use the same id of the project and it kinda worked well however after a few triies I lead to this problem. Like the projectid isnt the same has the ssh_key_id.
